packages/destination-actions/src/destinations/liveramp-audiences/audienceEntered/sftp.ts

```diff
@@ -0,0 +1,63 @@
+import { InvalidAuthenticationError } from '@segment/actions-core'
+import Client from 'ssh2-sftp-client'
+import path from 'path'
+import { Settings } from '../generated-types'
+
+const LIVERAMP_SFTP_SERVER = 'files.liveramp.com'
+const LIVERAMP_SFTP_PORT = 22
+const sftp = new Client()
+
+function validateSFTP(settings: Settings) {
```

Comment:
Consider DRYing this function
```suggestion
function validateSFTP(settings: Settings) {
  const rootErrorMessage = 'Selected SFTP upload mode, but missing '
  const requiredFields = [
    'sftp_username',
    'sftp_password', 
    'sftp_folder_path'
  ]

  requiredFields.forEach((field) => {
    if (!settings[field]) {
      throw new InvalidAuthenticationError(
        `${rootErrorMessage}${field.replace('_', ' ')}.`
      )
    }
  })
}
```

Reply:
I'd argue in favor of readability for this particular one.

Reply:
works for me!